packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/Drawing/DrawingContainer.js

```diff
@@ -0,0 +1,142 @@
+import cuid from 'cuid'
+import { inject, observer } from 'mobx-react'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+import getDrawingTool from './helpers/getDrawingTool'
+
+function storeMapper (stores) {
+  const { activeDrawingTool, eventStream } = stores.classifierStore.drawing
+  const { activeStepTasks } = stores.classifierStore.workflowSteps
+  return {
+    activeDrawingTool,
+    activeStepTasks,
+    eventStream
+  }
+}
+
+@inject(storeMapper)
+@observer
+class DrawingContainer extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      activeMark: null,
+      marks: new Map()
+    }
+
+    // TEMP CODE UNTIL COORDINATE STREAM
+    this.tempSubscription = null
+
+    this.finishMark = this.finishMark.bind(this)
+  }
+
+  componentDidMount () {
+    this.setActiveMark()
+
+    // TEMP CODE UNTIL COORDINATE STREAM
+    const { eventStream } = this.props
+    this.tempSubscription = eventStream.subscribe(event => {
+      if (event.type === 'pointerup') {
+        this.tempMarkCreation()
+      }
+    })
+  }
+
+  componentDidUpdate (prevProps) {
+    if (this.props.activeStepTasks !== prevProps.activeStepTasks || this.props.activeDrawingTool !== prevProps.activeDrawingTool) {
+      this.setActiveMark()
+    }
+  }
+
+  // TEMP CODE UNTIL COORDINATE STREAM
+  componentWillUnmount () {
+    if (this.tempSubscription) {
+      this.tempSubscription.unsubscribe()
+    }
+  }
+
+  // TEMP CODE UNTIL COORDINATE STREAM
+  tempMarkCreation () {
+    const { activeDrawingTool, activeStepTasks } = this.props
+
+    const [activeDrawingTask] = activeStepTasks.filter(task => task.type === 'drawing')
+    const tool = activeDrawingTask.tools[activeDrawingTool]
+    if (tool.type === 'line') {
+      const coordinates = { x1: (Math.floor(Math.random() * 400)), y1: (Math.floor(Math.random() * 400)), x2: (Math.floor(Math.random() * 400)), y2: (Math.floor(Math.random() * 400)) }
+      this.finishMark(coordinates)
+    }
+    if (tool.type === 'point') {
+      const coordinates = { x: (Math.floor(Math.random() * 400)), y: (Math.floor(Math.random() * 400)) }
+      this.finishMark(coordinates)
+    }
+  }
+
+  setActiveMark () {
+    const { activeDrawingTool, activeStepTasks } = this.props
+    const [activeDrawingTask] = activeStepTasks.filter(task => task.type === 'drawing')
```

Comment:
I think we discussed having the drawing store filter this on a view getter function? Totally ok if that's going to come in a later PR.

Reply:
Yes and yes, those changes are in PR #1164, [here's a link specifically to those views in the store](https://github.com/zooniverse/front-end-monorepo/pull/1164/files#diff-e76319fbfa25691959816b6c45adbf06R11).